src/app/shared/components/template/services/template-calc.service.ts

```diff
@@ -57,11 +65,20 @@ export class TemplateCalcService extends AsyncServiceBase {
       calc: (v: any) => v, // include simple function so @calc(...) returns the value already parsed inside
       app_day: this.dataEvaluationService.data.app_day,
       app_first_launch: this.dataEvaluationService.data.first_app_launch,
-      app_user_id: this.serverService.app_user_id,
-      device_info: this.serverService.device_info,
+      app_user_id: this.app_user_id,
+      device_info: this.device_info,
     };
   }
 
+  private async setUserMetaData() {
+    if (!this.device_info) {
+      this.device_info = await Device.getInfo();
+    }
+    if (!this.app_user_id) {
+      this.app_user_id = this.localStorageService.getProtected("APP_USER_ID");
+    }
```

Comment:
nit(non-blocking): As this is called on init (and not re-triggered) the `if` blocks aren't necessary. In fact might be good to remove so that the service can update the value if the user changes (e.g. through user import action) - not sure if we've covered this edge case already or not (guessing not).